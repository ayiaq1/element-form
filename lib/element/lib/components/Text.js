define(['require', 'exports', 'react', 'antd'], function (
  require,
  exports,
  react_1,
  antd_1,
) {
  'use strict';
  Object.defineProperty(exports, '__esModule', { value: true });
  var Text = function (_a) {
    var value = _a.value,
      _b = _a.ellipsis,
      ellipsis = _b === void 0 ? true : _b,
      style = _a.style;
    return react_1.default.createElement(
      antd_1.Typography.Text,
      {
        ellipsis: ellipsis
          ? {
              tooltip: ellipsis,
            }
          : false,
        style: style,
      },
      value,
    );
  };
  exports.default = (0, react_1.memo)(Text);
});
//# sourceMappingURL=Text.js.map
