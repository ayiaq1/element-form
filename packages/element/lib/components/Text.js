/*
 * @Author: yihuang
 * @Date: 2021-12-01 14:07:25
 * @Description: 只读文本，超长省略
 * @LastModifiedBy: yihuang
 */
import React, { memo } from 'react';
import { Typography } from 'antd';
var Text = function (_a) {
  var value = _a.value,
    fieldProps = _a.fieldProps;
  var _b = fieldProps !== null && fieldProps !== void 0 ? fieldProps : {},
    _c = _b.ellipsis,
    ellipsis = _c === void 0 ? true : _c,
    style = _b.style,
    className = _b.className;
  return React.createElement(
    Typography.Text,
    {
      ellipsis: ellipsis
        ? {
            tooltip: ellipsis,
          }
        : false,
      style: style,
      className: className,
    },
    value,
  );
};
export default memo(Text);
