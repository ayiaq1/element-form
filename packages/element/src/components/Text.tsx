/*
 * @Author: yihuang
 * @Date: 2021-12-01 14:07:25
 * @Description: 只读文本，超长省略
 * @LastModifiedBy: yihuang
 */
import React, { memo } from 'react';
import { Typography } from 'antd';
import type { ITextProps } from '../typings';

const Text = ({ value, ellipsis = true, style }: ITextProps) => (
  <Typography.Text
    ellipsis={
      ellipsis
        ? {
            tooltip: ellipsis,
          }
        : false
    }
    style={style}
  >
    {value}
  </Typography.Text>
);
export default memo(Text);
