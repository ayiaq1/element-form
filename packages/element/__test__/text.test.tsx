/*
 * @Author: yihuang
 * @Date: 2022-03-22 15:20:48
 * @Description: Text组件
 * @LastModifiedBy: yihuang
 */
import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import mountTest from '../tests/mountTest';
import Text from '../src/components/Text';

describe('Text', () => {
  mountTest(Text);

  /* Snapshot is a way to test the UI of your component. */
  it('Save Snapshot', async () => {
    const wrapper = renderer.create(<Text />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('default value', async () => {
    const value = 'default';
    const wrapper = mount(<Text value={value} />);
    expect(wrapper.text()).toEqual(value);
  });

  it('update value', async () => {
    const value = 'default';
    const wrapper = mount(<Text value="123" />);
    wrapper.setProps({ value });
    expect(wrapper.text()).toEqual(value);
  });
});
