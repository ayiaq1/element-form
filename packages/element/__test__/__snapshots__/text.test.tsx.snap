// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Save Snapshot 1`] = `
<span
  className="ant-typography ant-typography-ellipsis ant-typography-single-line"
  onClick={null}
  style={
    Object {
      "WebkitLineClamp": undefined,
    }
  }
>
  <span
    aria-hidden={true}
    style={
      Object {
        "display": "block",
        "left": 0,
        "pointerEvents": "none",
        "position": "fixed",
        "top": 0,
        "visibility": "hidden",
        "whiteSpace": "nowrap",
        "wordBreak": "keep-all",
        "zIndex": -9999,
      }
    }
  >
    lg
  </span>
  <span
    aria-hidden={true}
    style={
      Object {
        "display": "block",
        "left": 0,
        "margin": 0,
        "padding": 0,
        "pointerEvents": "none",
        "position": "fixed",
        "top": 0,
        "visibility": "hidden",
        "whiteSpace": "normal",
        "width": 0,
        "zIndex": -9999,
      }
    }
  >
    <span
      aria-hidden={true}
    >
      ...
    </span>
  </span>
</span>
`;
